core/src/main/scala/dagr/core/execsystem/GraphNode.scala

```diff
@@ -37,11 +39,10 @@ import scala.collection.mutable.ListBuffer
   *                         is used to track submission, start, and end dates for the parent node.  Typically parent nodes
   *                         are [[dagr.core.tasksystem.Pipeline]]s.
   */
-class GraphNode(val taskId: BigInt,
-                var task: Task,
+class GraphNode(var task: Task,
                 predecessorNodes: Traversable[GraphNode] = Nil,
                 var state: GraphNodeState.Value = GraphNodeState.PREDECESSORS_AND_UNEXPANDED,
-                val parent: Option[GraphNode] = None) {
+                val parent: Option[GraphNode] = None) extends BaseGraphNode {
 
   private val _predecessors = new ListBuffer[GraphNode]()
   private val _predecessorsStatic = new ListBuffer[GraphNode]()
```

Comment:
I see.  My tendency is nearly always to implement only what's needed now, and let future me figure out the best way to support future features, since it means that the system is as simple as it can be at any time.